components/SubmissionForm.vue

```diff
@@ -189,45 +204,64 @@ const validateFields = (e: Event) => {
     submissionStore.submit()
 }
 
-const submissionInputFields = [
-    { field: 'location', validate: validations.validateGoogleMapsUrlInput, validationReference: isValidGoogleMapsUrl },
-    { field: 'lastName', validate: validations.validateUserSubmittedLastName, validationReference: isValidLastName },
-    { field: 'firstName', validate: validations.validateUserSubmittedFirstName, validationReference: isValidFirstName },
-    { field: 'selectLanguage1', validate: validations.validateFirstSpokenLanguage, validationReference: isValidPrimarySpokenLanguage },
-    { field: 'selectLanguage2', validate: validations.validateSecondSpokenLanguage, validationReference: isValidSecondarySpokenLanguage }
-]
-
-const initialValidationCheck =
-async (inputValue: string, inputValidationCheck: (inputValue: string) => boolean, field: ValidationReference) => {
-    validationCheckedPreviously.value = true
-    await nextTick()
-    const validationReference = useCorrectValidation(field)
-    validationReference.value = inputValidationCheck(inputValue)
-}
-
-const useCorrectValidation = (field: ValidationReference) => {
+const initialValidationCheck = async (inputValue: string, field: string) => {
     switch (field) {
-        case ValidationReference.Location:
-            return isValidGoogleMapsUrl
-        case ValidationReference.LastName:
-            return isValidLastName
-        case ValidationReference.FirstName:
-            return isValidFirstName
-        case ValidationReference.PrimarySpokenLanguage:
-            return isValidPrimarySpokenLanguage
-        case ValidationReference.SecondarySpokenLanguage:
-            return isValidSecondarySpokenLanguage
+        case 'googleMaps':
+            googleMapsValidationCheckedPreviously.value = true
+            // nextTick is used to ensure that the model value is updated before doing the validation check
```

Comment:
What do you think about leaving this comment just once at the top of the switch/case?

Reply:
That sounds good